### PR TITLE
 #160451603 An Authenticated user should be able to comment on article

### PR DESCRIPTION
**Pull request by koiic**
#### What does this PR do?
- Create Comments
- Update Comments
   
#### Description of Task to be completed?
  - Implement create comment by authenticated user API ````POST /api/v1/articles/:slug/comments````
  - Implement update comment by auth user  API ````PUT /api/v1/articles/comment/:id````
  - Deny user's access to update other user's comment
  - Display comments when fetching Articles

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
NA

#### What are the relevant pivotal tracker stories?
[#160451603](https://www.pivotaltracker.com/story/show/160451603 "create and update comment")
 
### Question
 NA

### Screenshot
 NA

### CODE REVIEW

File: server/controllers/CommentController.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            message: 'Internal Server Error',
+          });
+        }
+        return response.status(201)
+          .json({
+            message: 'Comment Added Successfully',
+            data: { comment, user }
+          });
+      })
+      .catch(err => response.status(400).json({ message: err.message }));
```
Comment (Victor-Ugwueze) [inline server/controllers/CommentController.js]: I think the styles are a bit off, @koiic, please look through it.
 It should follow this style, check all occurrences.
```js 
  .catch(err => response.status(400)
         .json({
            message: err.message 
        }));
```

File: server/middlewares/CommentValidation.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   * @static
+   * @param {object} request
+   * @param {object} response
+   * @param {function} next
+   * @returns {object} object
+   * @memberof CommentValidation
+   */
+  static validateComment(request, response, next) {
+    const { commentBody } = request.body;
+    console.log('============>>>>', commentBody);
```
Comment (obulaworld) [inline server/middlewares/CommentValidation.js]: All console.logs should be removed before pushing to the repo such as line 20 which reads: 
`console.log('============>>>>', commentBody);`
A great way to do this is to use `command + f` buttons to search for and remove all occurrences of it.

**Review by obulaworld — CHANGES_REQUESTED** [review]

**Review by Victor-Ugwueze — CHANGES_REQUESTED** [review]

**Review by Victor-Ugwueze — APPROVED** [review]
LGTM

**Review by obulaworld — APPROVED** [review]
Looking Good To Me

File: server/controllers/CommentController.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+   * @description The create comment method
+   * @param  {object} request The request object
+   * @param  {object} response The response object
+   * @returns {object} json response
+   * @memberof CommentController
+   */
+  static addCommentToArticle(request, response) {
+    const user = request.userObject.dataValues;
+    const article = request.articleObject.dataValues;
+    // get the user id
```
Comment (KaiserPhemi) [inline server/controllers/CommentController.js]: #### Nitpick
Remove unnecessary comments

File: server/middlewares/fetchUser.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// modules import
+import model from '../db/models';
+
+const { User } = model;
+
+// Method to fetch the authenticated user
+const getUser = (request, response, next) => {
+  console.log(request.decoded);
```
Comment (KaiserPhemi) [inline server/middlewares/fetchUser.js]: #### Nitpick
Remove all instances of logging.

File: server/routes/api/comments.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// third-party libraries
```
Comment (KaiserPhemi) [inline server/routes/api/comments.js:1]: #### Nitpick
- Remove empty lines within imports
- Group imports appropriately

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Review by KaiserPhemi — APPROVED** [review]

**Comment by KaiserPhemi** [comment]
@koiic 
Please squash your commits.

**Pull request merged**